src/components/button/Button.tsx

```diff
@@ -104,7 +105,7 @@ const changeSeverity = (severity: string) => {
 };
 
 /* eslint-disable sonarjs/no-identical-functions */
-const StyledButton = styled.button<ButtonProps>`
+const StyledButton = styled.button<ButtonProps<React.ElementType>>`
```

Comment:
Do we need to provide `React.ElementType` here? default value "button" is defined.

Reply:
Make sense - adjusted in 8408605807413d8f9771ada7108ff00a2499dcd2